plugin.video.npogemist/resources/lib/npo_start.py

```diff
@@ -0,0 +1,483 @@
+        for image_function in image_functions:
+            thumb = image_function(episode, serie)
+            if thumb is not None:
+                return thumb
+        return ""
+
+    def _episode_image(self, episode, serie):
+        """Get the episode image."""
+        for format in ["original", "tv"]:
+            try:
+                image = episode["images"]["original"]["formats"][format]["source"]
+                if image is not None and image != "":
+                    return image
+            except (KeyError, TypeError):
+                pass
+        return None
+
+    def _series_image(self, episode, serie):
+        """Get the series' image."""
+        for format in ["original", "tv"]:
+            try:
+                image = serie["images"]["original"]["formats"][format]["source"]
+                if image is not None and image != "":
+                    return image
+            except (KeyError, TypeError):
+                pass
+        return None
+
+    def get_genres(self):
+        """Get a list of genres by detecting them from a lot of series."""
+        genres = []
+        catalogue = start_api_retriever.retrieve_series_random()
+        series = self._get_catalogue_series(catalogue)
+        if series is not None:
+            for serie in series:
+                if "genres" in serie:
+                    for genre in serie["genres"]:
+                        genres.extend(genre["terms"])
+        return sorted(set(genres))
+
+    def get_series_by_genre(self, genre):
+        """Get series by genre."""
+        catalogue = start_api_retriever.retrieve_series_genre(genre)
+        for component in catalogue["components"]:
+            if component["type"] == "grid":
+                for serie in component["data"]["items"]:
+                    if self._is_series(serie):
+                        yield self._serie2item(serie)
+
+    def get_search(self, search):
+        """get series and broadcasts by searching."""
+        # NPO doesn't like "/", even when encoded.
+        search_name = urllib.quote(search.replace("/", " "))
+        result = start_api_retriever.retrieve_items_by_search(search_name)
+        for item in result["items"]:
+            if self._is_series(item):
+                yield self._serie2item(item)
+            elif self._is_episode(item):
+                if self._is_not_expired(item["availability"]):
+                    yield self._episode2item(item)
+
+    def _is_episode(self, item):
+        """Return True if item is an episode."""
+        return item["type"] == "broadcast"
+
+    def get_subtitles_url(self, mid):
+        """Return a url for subtitles for this id."""
+        return "https://rs.poms.omroep.nl/v1/api/subtitles/{0}/nl_NL/CAPTION.vtt".format(mid)
+
+    def get_play_url(self, mid):
+        """Return a video url for this id."""
+        videodata = start_api_retriever.retrieve_video_data(mid)
+        return videodata['url']
+
+
+if __name__ == '__main__':
```

Comment:
looks like this is for local testing only, so please remove it from you submission